CodingChallenge/MusicSearch/MusicSearchInteractor.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  MusicSearchInteractor.swift
+//  CodingChallenge
+//
+//  Created by Edgar Barocio on 6/22/19.
+//  Copyright (c) 2019 Edgar Barocio. All rights reserved.
+//
+//  This file was generated by the Clean Swift Xcode Templates so
+//  you can apply clean architecture to your iOS and Mac projects,
+//  see http://clean-swift.com
+//
+
+import UIKit
+
+protocol MusicSearchBusinessLogic {
+    
+    func performSearch(request: MusicSearch.Search.Request)
+}
+
+protocol MusicSearchDataStore {
+    var songs: [SongResponse]? { get }
+}
+
+class MusicSearchInteractor: MusicSearchBusinessLogic, MusicSearchDataStore {
+    var presenter: MusicSearchPresentationLogic?
+    var searchWorker: MusicSearchWorker?
+    var songs: [SongResponse]?
+    
+    // MARK: Fetch songs
+    
+    func performSearch(request: MusicSearch.Search.Request) {
+        searchWorker = MusicSearchWorker()
```

Comment:
Why create a new searchWorker every time? Why not just give default initial value with the class?

Reply:
This is a good question, I need to look into a better implementation, this is the result of the clean swift templates but I did not take into account the worker would be re-created over and over